bundles/storage/org.eclipse.smarthome.storage.json/src/main/java/org/eclipse/smarthome/storage/json/JsonStorageService.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2014-2016 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.storage.json;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.smarthome.config.core.ConfigConstants;
+import org.eclipse.smarthome.core.storage.Storage;
+import org.eclipse.smarthome.core.storage.StorageService;
+import org.osgi.service.component.ComponentContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This implementation of {@link StorageService} provides a mechanism to store
+ * data in Json files.
+ *
+ * @author Chris Jackson - Initial Contribution
+ */
+public class JsonStorageService implements StorageService {
+
+    private final Logger logger = LoggerFactory.getLogger(JsonStorageService.class);
+
+    /** the folder name to store database ({@code jsondb} by default) */
+    private String dbFolderName = "jsondb";
+
+    private final String CFG_MAX_BACKUP_FILES = "backup_files";
+    private final String CFG_WRITE_DELAY = "write_delay";
+    private final String CFG_MAX_DEFER_DELAY = "max_defer_delay";
+
+    private int maxBackupFiles = 5;
+    private int writeDelay = 500;
+    private int maxDeferredPeriod = 60000;
+
+    private final Map<String, JsonStorage<Object>> storageList = new HashMap<String, JsonStorage<Object>>();
+
+    public void activate() {
```

Comment:
You have two activate methods now. You should move the code from here to the newly introduced one.